transformers does not detect Tensorflow when installing tensorflow-cpu package

### System Info

This issue was introduced in transformers 4.30.x.



Output of pip list (partial):

```
tensorflow-cpu                       2.12.0
tensorflow-estimator                 2.12.0
tensorflow-intel                     2.12.0
tensorflow-io-gcs-filesystem         0.31.0
transformers                         4.30.1
```

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

Manifests itself like this:

```
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
ImportError while loading 

<REDACTED>

    from transformers import file_utils, modeling_tf_outputs, modeling_tf_utils
.venv\lib\site-packages\transformers\modeling_tf_utils.py:42: in <module>
    from .generation import GenerationConfig, TFGenerationMixin
E   ImportError: cannot import name 'TFGenerationMixin' from 'transformers.generation' (<REDACTED>)
```

### Expected behavior

No errors, Tensorflow detected.

--- a/src/transformers/utils/import_utils.py
+++ b/src/transformers/utils/import_utils.py
@@ -40,6 +40,8 @@ logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
 # TODO: This doesn't work for all packages (`bs4`, `faiss`, etc.) Talk to Sylvain to see how to do with it better.
 def _is_package_available(pkg_name: str, return_version: bool = False) -> Union[Tuple[bool, str], bool]:
     # Check we're not importing a "pkg_name" directory somewhere but the actual library by trying to grab the version
+    # Note: _is_package_available("tensorflow") fails for tensorflow-cpu. Please test any changes to the line below
+    # with tensorflow-cpu to make sure it still works!
     package_exists = importlib.util.find_spec(pkg_name) is not None
     package_version = "N/A"
     if package_exists:

--- a/src/transformers/utils/import_utils.py
+++ b/src/transformers/utils/import_utils.py
@@ -147,7 +147,7 @@ if FORCE_TF_AVAILABLE in ENV_VARS_TRUE_VALUES:
     _tf_available = True
 else:
     if USE_TF in ENV_VARS_TRUE_AND_AUTO_VALUES and USE_TORCH not in ENV_VARS_TRUE_VALUES:
-        _tf_available = _is_package_available("tensorflow")
+        _tf_available = importlib.util.find_spec("tensorflow") is not None
         if _tf_available:
             candidates = (
                 "tensorflow",

--- a/src/transformers/utils/import_utils.py
+++ b/src/transformers/utils/import_utils.py
@@ -40,8 +40,6 @@ logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
 # TODO: This doesn't work for all packages (`bs4`, `faiss`, etc.) Talk to Sylvain to see how to do with it better.
 def _is_package_available(pkg_name: str, return_version: bool = False) -> Union[Tuple[bool, str], bool]:
     # Check we're not importing a "pkg_name" directory somewhere but the actual library by trying to grab the version
-    # Note: _is_package_available("tensorflow") fails for tensorflow-cpu. Please test any changes to the line below
-    # with tensorflow-cpu to make sure it still works!
     package_exists = importlib.util.find_spec(pkg_name) is not None
     package_version = "N/A"
     if package_exists:
@@ -149,6 +147,8 @@ if FORCE_TF_AVAILABLE in ENV_VARS_TRUE_VALUES:
     _tf_available = True
 else:
     if USE_TF in ENV_VARS_TRUE_AND_AUTO_VALUES and USE_TORCH not in ENV_VARS_TRUE_VALUES:
+        # Note: _is_package_available("tensorflow") fails for tensorflow-cpu. Please test any changes to the line below
+        # with tensorflow-cpu to make sure it still works!
         _tf_available = importlib.util.find_spec("tensorflow") is not None
         if _tf_available:
             candidates = (

